clients/fossid-webapp/src/main/kotlin/Extensions.kt

```diff
@@ -32,11 +29,13 @@ private const val SCAN_GROUP = "scans"
 private const val PROJECT_GROUP = "projects"
 
 /**
- * This function verifies that a request to FossID server was successful
+ * This function verifies that a request to FossID server was successful.
  * @param operation the label of the operation, for logging
  * @param withDataCheck if true, the payload will be checked for existence
  */
-fun EntityPostResponseBody<*>.checkResponse(operation: String, withDataCheck: Boolean = true) {
+fun <B, T> B?.checkResponse(operation: String, withDataCheck: Boolean = true):
+        B where B : EntityPostResponseBody<T> {
+    requireNotNull(this)
```

Comment:
Instead of requiring `this` to be not `null` here, why not simply defining this to be an extension function on non-nullable `B` instead of `B?`?

Reply:
Because this function is there to enforce non-nullability.
If the receiver was non-nullable, I would have to move the check outside on each call, defeating the purpose of this function.